### PR TITLE
refactor: laboratory wagmi tests

### PR DESCRIPTION
**Pull request by zoruka**
# Description

This PR is reducing the amount of RPC calls made when accessing wagmi laboratory test page by only triggering them when the test is triggered. It also reduce the usage of wagmi hooks when the wallet is not connected or the pre-conditions for starting the test is not met. Alongside with these changes, some logic replicated between the test components is moved into a custom hook to be reused across the components.  

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

APKT-684

# Showcase (Optional)

A live version may be seen [here](https://web3modal-laboratory-git-refactor-laborat-c937ae-walletconnect1.vercel.app/library/wagmi/).

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections

### CODE REVIEW

File: apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx [hunk 1]
```diff
@@ -1,86 +1,71 @@
...
-    isLoading,
-    isFetched,
-    error,
-    data: callsStatusResult
-  } = useCallsStatus({
+  const { isLoading, refetch: getCallsStatus } = useCallsStatus({
     id: batchCallId,
     query: {
-      enabled: enableGetCallsStatus && isConnected
+      enabled: false
```
Comment (tomiir) [inline apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx:46]: is this ok ?  cc @KannuSingh

File: apps/laboratory/src/components/Wagmi/WagmiWriteContractTest.tsx [hunk 2]
```diff
@@ -18,30 +31,41 @@ export function WagmiWriteContractTest() {
...
+  } = useSimulateContract({
     abi,
     address,
     functionName: 'purchase',
     value: parseEther('0.0001'),
-    args: [1]
+    args: [1],
+    query: {
+      enabled: false
+    }
```
Comment (tomiir) [inline apps/laboratory/src/components/Wagmi/WagmiWriteContractTest.tsx:51]: same

**Review by tomiir — APPROVED** [review]

File: apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx (hunk 1 above)
Comment (zoruka) [reply apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx:46]: The idea is to disable the fetch on render and be triggered only by the `refetch` function

Comment (KannuSingh) [reply apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx:46]: Looks good.

Comment (KannuSingh) [reply apps/laboratory/src/components/Wagmi/WagmiGetCallsStatusTest.tsx:46]: Can you just check if this not getting triggered on empty batchId?

**Pull request merged**